hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java

```diff
@@ -2458,6 +2443,48 @@ public FileArchiveNotificationResponse reportFileArchival(RpcController controll
  
   // HBCK Services
 
+  @Override
+  public MergeTableRegionsResponse mergeRegions(RpcController c, MergeTableRegionsRequest request)
```

Comment:
Thanks for your comment @wchevreuil . But shouldn't we be able to determine whether the request is coming in from hbck or regular client? because we want to do some extra steps in case of non-hbck generated request. Like calling cp hooks? or master initialization check? and our most recent HBASE-23864 which prevents us from submitting a merge regions procedure in case the split/merge switch is turned off? We wouldn't want hbck to adhere to split/merge switch rules right?

Reply:
> But shouldn't we be able to determine whether the request is coming in from hbck or regular client?

Yeah, that's the main question. My view is that the use case for both are the same, users triggering merges via shell or java API are already performing some sort of advanced admin operation, if those fail then they would just resort to hbck. I think a bit confusing that we provide two methods to do the same thing, only different in the level of extra checks performed.

Reply:
Can we just add a boolean flag to MergeTableRegionsRequest?

Reply:
I like the @infraio  suggestion.

Or have a queryable client identity?